### PR TITLE
Emit Sidebar events on nextcloud-bus

### PR DESCRIPTION
**Pull request by artonge**
Emit sidebar events on nextcloud's event bus.

The only tweak is for the `closed` event. As commented:

> For the `handleClosed` event listener, we need to listen to the `hook:destroyed` event instead of the `closed` event.
> This is necessary because with v-if, the 'closed' event is emitted after the destruction of AppSidebar, and all the event listeners are unbound and won't be triggered.

This assumed that the `handleClosed` event occurres almost at the same time as the `closed` event.

### CODE REVIEW

File: apps/files/src/views/Sidebar.vue [hunk 1]
```diff
@@ -29,7 +33,11 @@
 		@close="close"
 		@update:active="setActiveTab"
 		@update:starred="toggleStarred"
-		@[defaultActionListener].stop.prevent="onDefaultAction">
+		@[defaultActionListener].stop.prevent="onDefaultAction"
+		@opening="handleOpening"
+		@opened="handleOpened"
+		@closing="handleClosing"
+		@hook:destroyed="handleClosed">
```
Comment (skjnldsv) [inline apps/files/src/views/Sidebar.vue:40]: Ah, that is interesting! :thinking: 
Shouldn't that be handled in the vue components? Is there something we can do there?
Emit on beforeDestroy? I assume the animation will be done at this moment?

Comment (artonge) [reply apps/files/src/views/Sidebar.vue:40]: I thought about making changes to the `AppSidebar` component, but the problem comes from how it is used and not from its implementation, so I think it makes more sense to circumvent the limitation of `v-if` here than inside the component.

Comment (skjnldsv) [reply apps/files/src/views/Sidebar.vue:40]: We could do both?
Cover `show` and `if` ?

**Review by skjnldsv — APPROVED** [review]

**Comment by skjnldsv** [comment]
Needs `@nexxtcloud/vue@3.10.1`

**Comment by skjnldsv** [comment]
Tests fails, please wait for https://github.com/nextcloud/server/pull/27265 before rebasing

**Comment by blizzz** [comment]
> Node tests fails, please wait for #27265 before rebasing

is merged

**Comment by MorrisJobke** [comment]
Conflicts :/

**Comment by skjnldsv** [comment]
Please rebase @artonge

**Comment by artonge** [comment]
All is green !

**Review by MorrisJobke — APPROVED** [review]
Change looks good 👍

**Pull request merged**